### DISCUSSION TITLE
Commemoration with a ferial office

### DISCUSSION THREAD

**Discussion by Augustinus-Altovadensis** [DivinumOfficium/divinum-officium#2662] [Developers corner]
Laudétur Jesus Christus!

I started to work on calendar in the Cistercian office and I've come to a curious stop:
- in the Cistercian Office, there is a (quite frequent) option to have a ferial office all with the Collect from Sunday, while adding a commemoration of a (obviously) minor Saint. 

Now my question: **how can I do this in the Sancti/XX-XX.txt file?**

When using "vide" or "ex" directive, it changes the Chapter and all further part of the Office. When I omit the "ex/vide" directive, I get no Collect, the rest is completely fine.  Is there any way to get the Collect from Sunday without changing the code? 

**Thanks!**

**Comment by ajmalton** [comment]
I think the answer is Yes.  But I'm not perfectly sure what the requirement is.

On these tempora ferial days (rank 1?) there's a occurrent kalendar day which should be commemorated -- when? at Mass and the major hours?

And otherwise the office is wholly ferial?

**Reply by Augustinus-Altovadensis** [reply]
I would find it logical too, that the answer is yes, I just don't know, _how_ to do it. And I'm afraid I will have to change the code too, because the current implementation of the Monastic rite is quite light on Commemorations and they leave out many of perfectly legal options, at least for the Cistercian rite. And this is another huge can of worms, not only is the code not really simple (which is understandable, as the rules aren't either), but there are Cistercian parts that I will need to add too. Like the **Commemoratio** B.M.V. for all the feasts up to xij. Lect. et M. And yes, it is a Commemoration, not a Suffragium or Marian Antiphon. And apart from that, there are also Commemorations of De S. Joseph, De SS. Benedicto, Bernardo et Roberto, and De Pace. These are not really crucial, but they are a firm part of the Cistercian Office, although currently long gone...

There are two ranks of feasts of Saints, when the ferial office is used: 
- the lowest rank "Commemoratio" only commemorates the feast on top of a ferial Office (first Vespers and Lauds) in the ferial Mass. 
- the second lowest rank "Commemoratio et Missa" commemorates the feast on top of a ferial Office (first Vespers and Lauds), but the Mass is from the feast.
Then also the third lowest rank "iij. Lectiones et Missa" utilizes the ferial Psalms, but the rest of the Office is from the Common.

Unlike the Roman Office, every feast in the Cistercian Office has the first Vespers, but only the 12-Lesson feasts have the second Vesper on the day of the feast.

And to answer your question: **yes, the office is wholly ferial**, Psalms, Antiphons, Chapter, Responsory, Hymn, Versicle, everything.

BTW, if you have some pointers, what to change and where, I'll be glad for any kind of help. The redundancy of the code and some parts abandoned for the time being don't really make this easy either. Thanks!

**Comment by ajmalton** [comment]
Ok, now I think I understand, and I not at all sure if it can be done without changing the code.

You need commemorations, not suffrage: that means (am I right?) that those commemorations you mentioned occur on the _Saints’ dates_, but with the office still ferial? Not like a suffrage which is prayed on certain days based on the rank or other character of the day _itself_ (eg penitential)?

Is it otherwise a normal commemoration? What about the order of multiple commemorations? (done by the Roman rules currently.)

The general approach in the code is to prepare commemorations but then erase them if there's a rule saying so.  Black list rather than white list, so to speak.  So what happens now in the Monastic version if you set up the commemorations you want?  Can we see where in the code they are being "erased"?  Somewhere in horascommon lines 757 and following?

Or perhaps the Saints’ days "win" and the your feriae are being ignored?  If so, then what if you sightly increase the rank number of the feriae? So that they look a bit like privilegiatae for this purpose? (Monastic only, of course!)

These are only ideas to try.
